### PR TITLE
convert markdown files to rst

### LINKED ISSUE #453: [doc/changelog] link objects to api doc

### PR DESCRIPTION
**Pull request by marscher**
## Changes

- The changelog and readme have been converted to rst. The changelog now properly links all object within Weldx and further externel ones.
- Improved docstrings for WeldxFile
- fixed doc generation for weldx.util

## Related Issues

Closes #453, #508

## Checks

- [x] updated CHANGELOG.md
- [x] updated doc/

### CODE REVIEW

File: changelog.rst [hunk 1]
```diff
@@ -0,0 +1,970 @@
...
+deprecations
+~~~~~~~~~~~~
+
+dependencies
+~~~~~~~~~~~~
+
+0.4.1 (20.07.2021)
+------------------
+
+.. _added-1:
```
Comment (CagtayFabry) [inline changelog.rst:168]: does something break if we delete these links?

**Comment by CagtayFabry** [comment]
I think the rst syntax looks also rather easy to update manually 👍

File: changelog.rst (hunk 1 above)
Comment (marscher) [reply changelog.rst]: dont think so. I've removed them

**Comment by marscher** [comment]
This ready for review. I think the linting things are false positives.

File: doc/conf.py [hunk 2]
```diff
@@ -26,18 +26,18 @@
...
-    import sphinx.builders.html
-    import sphinx.builders.latex
-    import sphinx.builders.texinfo
-    import sphinx.builders.text
-    import sphinx.ext.autodoc
+    import sphinx.builders.html  # noqa
+    import sphinx.builders.latex  # noqa
+    import sphinx.builders.texinfo  # noqa
+    import sphinx.builders.text  # noqa
+    import sphinx.ext.autodoc  # noqa
```
Comment (vhirtham) [inline doc/conf.py:40]: just curious: Which warnings are suppressed here?

**Review by vhirtham — APPROVED** [review]

**Comment by vhirtham** [comment]
Didn't check the converted files for correctness. Just had a short glimpse. I trust the tool to get it done. However, you should quickly double-check all touched pages on RTD. The main page has a duplicate caption for example.

File: doc/conf.py (hunk 2 above)
Comment (marscher) [reply doc/conf.py:40]: unused imports, the new pre-commit tool would remove them without the noqa. Then sphinx would break.

**Comment by marscher** [comment]
> Didn't check the converted files for correctness. Just had a short glimpse. I trust the tool to get it done. However, you should quickly double-check all touched pages on RTD. The main page has a duplicate caption for example.

The changelog was checked intensively. Good catch with the duplicated headline. I will fix it.

**Pull request merged**